idea/src/org/jetbrains/kotlin/idea/versions/KotlinRuntimeLibraryUtil.java

```diff
@@ -287,16 +314,24 @@ static void replaceFile(File updatedFile, VirtualFile replacedJarFile) {
     @NotNull
     private static Collection<VirtualFile> getLibraryRootsWithAbiIncompatibleVersion(
             @NotNull Project project,
-            @NotNull ScalarIndexExtension<Integer> index
+            @NotNull ScalarIndexExtension<Integer> index,
+            @NotNull Predicate<Module> checkModule,
+            @NotNull Predicate<Integer> checkAbiVersion
     ) {
         ID<Integer, Void> id = index.getName();
-        Collection<Integer> abiVersions = FileBasedIndex.getInstance().getAllKeys(id, project);
-        Set<Integer> badAbiVersions = Sets.newHashSet(Collections2.filter(abiVersions, new Predicate<Integer>() {
-            @Override
-            public boolean apply(Integer abiVersion) {
-                return !AbiVersionUtil.isAbiVersionCompatible(abiVersion);
+
+        Module[] modules = ModuleManager.getInstance(project).getModules();
+
+        List<Module> modulesToCheck = new ArrayList<Module>();
+        for(Module module : modules) {
```

Comment:
If you're using `Predicate`, you might as well go ahead and call `Collections2.filter` here. Also I would recommend to use `kotlin.Function1` and `KotlinPackage.filter` instead, but that's up to you